server/v2/core/appmanager/app.go

```diff
@@ -16,6 +16,9 @@ import (
 // In the block building phase the transactions from the mempool can be verified, executed right away or lazily.
 type PrepareHandler[T transaction.Tx] func(context.Context, uint32, mempool.Mempool[T], store.ReadonlyState) ([]T, error)
 
+// ProcessHandler is a function that takes a list of transactions and returns a boolean and an error. If the verification of a transaction fails, the boolean is false and the error is non-nil.
```

Comment:
Do we really need a boolean? Why not just `error`. The caller can infer success if the error is nil or not.